lua/nvim-tree/git/cli.lua

```diff
@@ -0,0 +1,136 @@
+local notify = require "nvim-tree.notify"
+local log = require "nvim-tree.log"
+-- vim.loop got renamed in 0.10.0
+-- ref: https://github.com/neovim/neovim/pull/22846
+local uv = vim.loop or vim.uv
+
+local M = {}
+
+-- Loggers
+M.notify = notify
+M.log = log
+
+--- See h `uv.spawn-options` for full list of props
+--- @class vim.uv.SpawnOptions
+--- @field args string[] args to be passed to git command e.g. -C
+--- @field env table|nil
+--- @field cwd table|nil
+--- @field detached boolean|nil detach process from parent
+--- @field timeout number|nil timeout before hanged git process gets killed
+
+--- A very lightweight generic git cli command wrapper powered by libuv;
+--- it's probably worth to consider to import another library
+--- @param uv_spawn_opts vim.uv.SpawnOptions uv.spaw like options
+--- @param on_err_or_data fun(err: string|nil, data: string|nil) error handler callback
+--- @param on_exit? fun(code: number, signal: number) error handler callback
+--- @param on_timeout? fun(): nil called if process timedout
+function M.cli(uv_spawn_opts, on_err_or_data, on_exit, on_timeout)
+  if (not M._GIT_IN_PATH) and vim.fn.executable "git" == 0 then
+    M.notify.error "nvim-tree: git not in path. Install git first. Aborting."
+    return
+  end
+  M._GIT_IN_PATH = true
+
+  if type(uv_spawn_opts) ~= "table" then
+    error("uv_spawn_opts table is required", 2)
```

Comment:
Throwing errors like this is not acceptable. This is not public API so should fail gracefully and silently.

In the case of an unrecoverable error, `disable_git_integration` should be called to inform the user and disable git integration.

Reply:
@alex-courtis This error is for developer to ensure function is properly called by wrapper functions, not for enduser. It has nothing to do with git subprocess errors which are always handled properly.

Reply:
nvim-tree does not check arguments in this manner, we ensure, via annotations etc. and manual confirmation that callers pass the correct arguments.

That is the code style, please follow it.

Reply:
@alex-courtis Sorry but this is dumb. If we don't check input, we risk becoming internal consumers of the arguments  error-ridden. 

This is not how mature programming is done lol.

I will remove them in next revision for now, but it's really dumb and heavy nitpcking.